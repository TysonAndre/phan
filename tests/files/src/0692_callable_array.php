<?php

namespace TestNarrowingArray;

/**
 * @param ?callable-array $o
 */
function test($o) {
    if (is_array($o)) {
        $o();
        echo count($o);
        echo intdiv($o, 2);
    } else {
        echo intdiv($o, 2);
    }
}

function test_callable_object(callable $c, array $arr) {
    if (is_array($c)) {
        echo intdiv($c, 2);
        $c();
        assert(!empty($c));
    }

    if (is_callable($arr)) {
        echo intdiv($arr, 2);
        $arr();
        assert(!empty($arr));
    }
}
