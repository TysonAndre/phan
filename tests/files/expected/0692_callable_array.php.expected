%s:12 PhanTypeMismatchArgumentInternal%SReal Argument 1 ($num1) is $o of type callable-array (real type array) but \intdiv() takes int
%s:14 PhanTypeMismatchArgumentInternal Argument 1 ($num1) is $o of type null but \intdiv() takes int
%s:20 PhanTypeMismatchArgumentInternal%SReal Argument 1 ($num1) is $c of type callable-array but \intdiv() takes int
%s:22 PhanImpossibleCondition Impossible attempt to cast $c of type callable-array to empty
%s:26 PhanTypeMismatchArgumentInternal%SReal Argument 1 ($num1) is $arr of type callable-array but \intdiv() takes int
%s:28 PhanImpossibleCondition Impossible attempt to cast $arr of type callable-array to empty
